test/cases/adapters/postgresql/virtual_column_test.rb

```diff
@@ -3,18 +3,37 @@
 require "cases/helper"
 require "support/schema_dumping_helper"
 
-if ActiveRecord::Base.connection.supports_virtual_columns?
-  class PostgresqlVirtualColumnTest < ActiveRecord::PostgreSQLTestCase
-    include SchemaDumpingHelper
+module CockroachDB
+  if ActiveRecord::Base.connection.supports_virtual_columns?
+    class PostgresqlVirtualColumnTest < ActiveRecord::PostgreSQLTestCase
+      include SchemaDumpingHelper
 
-    self.use_transactional_tests = false
+      self.use_transactional_tests = false
 
-    def test_schema_dumping
-      output = dump_table_schema("virtual_columns")
-      assert_match(/t\.virtual\s+"upper_name",\s+type: :string,\s+as: nil, stored: true$/i, output)
-      assert_match(/t\.virtual\s+"name_length",\s+type: :integer,\s+as: "length\(\(name\)::text\)", stored: true$/i, output)
-      assert_match(/t\.virtual\s+"name_octet_length",\s+type: :integer,\s+as: "octet_length\(\(name\)::text\)", stored: true$/i, output)
-      assert_match(/t\.virtual\s+"column2",\s+type: :integer,\s+as: "\(column1 \+ 1\)", stored: true$/i, output)
+      class VirtualColumn < ActiveRecord::Base
+      end
+
+      def setup
+        @connection = ActiveRecord::Base.connection
+        @connection.create_table :virtual_columns, force: true do |t|
+          t.string  :name
+          t.virtual :upper_name,  type: :string,  as: "UPPER(name)", stored: true
+          t.virtual :name_length, type: :integer, as: "LENGTH(name)", stored: true
+          t.virtual :name_octet_length, type: :integer, as: "OCTET_LENGTH(name)", stored: true
+          t.integer :column1
+          t.virtual :column2, type: :integer, as: "column1 + 1", stored: true
+        end
+        VirtualColumn.create(name: "Rails")
+      end
+
+      # TODO: is this test result acceptable?
```

Comment:
Here's the CRDB dump
```ruby
  create_table "virtual_columns", force: :cascade do |t|
    t.string "name"
    t.virtual "upper_name", type: :string, as: "upper(name)", stored: true
    t.virtual "name_length", type: :bigint, as: "length(name)", stored: true
    t.virtual "name_octet_length", type: :bigint, as: "octet_length(name)", stored: true
    t.bigint "column1"
    t.virtual "column2", type: :bigint, as: nil, stored: true
  end
```

And here's the original pg expectation:
```ruby
t.virtual "upper_name", type: :string, as: "upper((name)::text)", stored: true
t.virtual "name_length", type: :integer, as: "length((name)::text)", stored: true
t.virtual "name_octet_length", type: :integer, as: "octet_length((name)::text)", stored: true
t.virtual "column2", type: :integer, as: "(column1 + 1)", stored: true
```

Does the current dump make sense to you @rafiss ?

Reply:
this one does seem like a CRDB bug. are you able to track down where the `as` field gets populated?

Reply:
Will look for it for sure. What about the other changes ? The `::text` mainly, since the `bigint` I think is ok as we already discussed that the naming and defaults of numbers are a bit different from pg

Reply:
The `as` population is found and fixed in the latest commit !